internal/infrastructure/kubernetes/deployment.go

```diff
@@ -250,13 +252,16 @@ func (i *Infra) createDeployment(ctx context.Context, infra *ir.Infra) (*appsv1.
 //
 // TODO: Update k/v pair to use gatewayclass controller name to distinguish between
 //       multiple Envoy Gateways.
-func EnvoyPodSelector() *metav1.LabelSelector {
+func EnvoyPodSelector(gatewayName string) *metav1.LabelSelector {
 	return &metav1.LabelSelector{
-		MatchLabels: envoyLabels(),
+		MatchLabels: envoyLabels(gatewayName),
 	}
 }
 
 // envoyLabels returns the labels used for Envoy.
-func envoyLabels() map[string]string {
-	return map[string]string{"app": "envoy"}
+func envoyLabels(gatewayName string) map[string]string {
```

Comment:
The input name should be `className`, `gcName`, or something that represents a GatewayClass and not Gateway b/c the ProxyInfra name is established from the managed GatewayClass.

xref: https://github.com/envoyproxy/gateway/blob/v0.2.0-rc1/internal/gatewayapi/translator.go#L97